New_Authenticator.ipynb

```diff
@@ -0,0 +1,71 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "from jupyterhub.auth import Authenticator\n",
+    "\n",
+    "from tornado import gen\n",
+    "\n",
+    "import csv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   
+   "outputs": [],
+   "source": [
+    "class MyNewAuthenticator(Authenticator):\n",
+    "        \n",
+    "        \n",
+    "        \n",
+    "\n",
+    "    @gen.coroutine\n",
+    "    def authenticate(self, handler,data):\n",
+    "       with open('Example.csv') as csvfile:\n",
+    "        readCSV=csv.reader(csvfile,delimiter=',')\n",
+    "        \n",
+    "        for row in readCSV:\n",
+    "            username=row[0]\n",
+    "            password=row[1]\n",
+    "            \n",
+    "            \n",
+    "            if username == data['username']:\n",
+    "                if password == data['password']:\n",
+    "                    return username\n",
+    "                return None\n",
+    "            return None\n",
```

Comment:
What will happen here if the username doesn't match the first row, but does match the second?